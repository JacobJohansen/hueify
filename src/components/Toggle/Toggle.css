.toggle-button {
  width: 30px;
  height: 18px;
  display: inline-block;
  vertical-align: middle;
  margin-top: -3px;
  cursor: pointer;
  border: solid 2px var(--grey);
  border-radius: 16px;
  transition: all .2s;
}

.toggle-button:before {
  content: ' ';
  display: inline-block;
  position: relative;
  top: -2px;
  left: 1px;
  width: 12px;
  height: 12px;
  border-radius: 50%;
  background: var(--grey);
  transition: all .2s;
}

.toggle-button.active {
  border-color: var(--blue);
}

.toggle-button.active:before {
  content: '';
  transform: translateX(12px);
  background: var(--blue);
}
