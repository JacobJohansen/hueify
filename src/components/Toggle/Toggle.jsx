import './Toggle.css'

export default ({ checked = false, onClick, style }) => (
  <div
    style={style}
    className={['toggle-button', checked ? 'active' : ''].join(' ')}
    onClick={onClick}
  />
)
