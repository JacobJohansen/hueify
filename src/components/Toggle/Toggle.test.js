/* eslint-env jest */
import { h, render } from 'preact'
import { default as renderToString } from 'preact-render-to-string'
import Toggle from './Toggle'

global.h = h

test('Should render toggle button', () => {
  const vdom = renderToString(<Toggle />)
  expect(vdom).toMatchSnapshot()
})

test('Should render checked toggle button', () => {
  const vdom = renderToString(<Toggle checked />)
  expect(vdom).toMatchSnapshot()
})

test('Should style component', () => {
  const vdom = renderToString(<Toggle style={{ backgorund: 'green' }} />)
  expect(vdom).toMatchSnapshot()
})

test('Should fire onClick event', done => {
  const onClick = () => done()

  render(
    <Toggle onClick={onClick} />,
    document.body
  )

  document.querySelector('.toggle-button').click()
})
