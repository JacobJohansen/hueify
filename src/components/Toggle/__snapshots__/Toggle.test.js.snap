// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render checked toggle button 1`] = `"<div class=\\"toggle-button active\\"></div>"`;

exports[`Should render toggle button 1`] = `"<div class=\\"toggle-button \\"></div>"`;

exports[`Should style component 1`] = `"<div style=\\"backgorund: green;\\" class=\\"toggle-button \\"></div>"`;
